processor/schemaprocessor/internal/migrate/signal.go

```diff
@@ -0,0 +1,97 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package migrate // import "github.com/open-telemetry/opentelemetry-collector-contrib/processor/schemaprocessor/internal/migrate"
+
+import "github.com/open-telemetry/opentelemetry-collector-contrib/processor/schemaprocessor/internal/alias"
+
+// SignalType allows for type constraints in order
+// to apply to potential type defined strings.
+type SignalType interface {
+	~string
+}
+
+// Signal allows for migrating types that
+// implement the `alias.Signal` interface.
+type Signal struct {
+	updates  map[string]string
+	rollback map[string]string
+}
+
+type SignalSlice []*Signal
+
+// NewSignal will create a `Signal` that will check the provided mappings if it can update a `alias.Signal`
+// and if no values are provided for `matches`, then all values will be updated.
+func NewSignal[Key SignalType, Value SignalType](mappings map[Key]Value) *Signal {
```

Comment:
Same here: do we have a case where SignalType is not a string?

Reply:
Yes, it is defined here `https://pkg.go.dev/go.opentelemetry.io/otel/schema@v0.0.4/v1.0/ast#MetricsChange` as `map[types.MetricName]types.MetricName` which are new types being defined instead of type alias being used.